src/predicate.ts

```diff
@@ -216,7 +229,9 @@ export function fieldFilterExpression(predicate: FieldPredicate, useInRange = tr
     const lower = predicate.gte;
     return `${fieldExpr}>=${predicateValueExpr(lower, timeUnit)}`;
   } else if (isFieldOneOfPredicate(predicate)) {
-    return 'indexof([' + predicateValuesExpr(predicate.oneOf, timeUnit).join(',') + '], ' + fieldExpr + ') !== -1';
+    return `indexof([${predicateValuesExpr(predicate.oneOf, timeUnit).join(',')}], ${fieldExpr}) !== -1`;
+  } else if (isFieldValidPredicate(predicate)) {
+    return predicate.valid ? `${fieldExpr}!==null&&!isNaN(${fieldExpr})` : `${fieldExpr}===null||isNaN(${fieldExpr})`;
```

Comment:
I'd prefer having spaces for readability.  

It might be worth submitting another PR to standardize other the filter expressions too.  
(Currently, some do have spaces between operators, some do not.)

Reply:
Let's do this in a follow up. Issue: https://github.com/vega/vega-lite/issues/4112